\documentclass[10pt, a4paper]{report}
\usepackage{authblk}
\usepackage{listings}
\usepackage{hyperref}

\title{NEXT Bioinformatics: Best Practices}

\author[1]{Rasmus Froberg Br\o ndum}
\author[2]{Michael Knudsen}
\author[3]{Mark Burton}
\author[4]{Savvas Kinalis}

\affil[1]{Department of Haematology, Aalborg University Hospital}
\affil[2]{MOMA}
\affil[3]{SDU}
\affil[4]{Rigshospitalet}

\begin{document}
\maketitle
\chapter{Introduction}
These workflows are based on the \href{https://software.broadinstitute.org/gatk/best-practices/}{GATK best practices}
The workflow requires specification of paths to a number of programs and reference datasets which must be downloaded and installed first.
\section{Programs}
\begin{lstlisting}[language=bash]
java="/usr/lib/jvm/jre-1.8.0-openjdk.x86_64/bin/java -Djava.io.tmpdir=$TMPDIR" 
gatk=/data/apps/software/GenomeAnalysisTK-3.6.jar
picard=/data/apps/software/picard-tools-2.0.1/picard.jar
\end{lstlisting}

\section{Datasets}
\begin{lstlisting}[language=bash]
assembly=/data/appdata/gatk_bundle_b37/human_g1k_v37.fa
known_snp=/data/appdata/gatk_bundle_b37/dbsnp_138.b37.vcf
known_indels=/data/appdata/gatk_bundle_b37/1000G_phase1.indels.b37.vcf
regions=/data/users/rasmus/agilent_sureselect_V5.bed
cosmic=/data/appdata/COSMIC/grch37/CosmicCodingMuts_v76_reordered.vcf
\end{lstlisting}


\chapter{Exome-seq}
\section{Alignment}

\begin{lstlisting}[language=bash]
## Align reads
/data/apps/bwa-0.7.12/bwa mem \
-M \
-t8 \
-R"@RG\tID:exome_1\tPL:ILLUMINA\tPU:exome.11\tLB:$id\tSM:$id\tCN:AAU" \
$assembly \
$rawDataDir/$R1 \
$rawDataDir/$R2 \
> aligned_reads.sam
\end{lstlisting}

\section{Marking duplicates}
\begin{lstlisting}[language=bash]
## Sort and convert to bam
$java -jar $picard SortSam \
INPUT=aligned_reads.sam \
OUTPUT=sorted_reads.bam \
SORT_ORDER=coordinate

## Mark PCR duplicates
$java -jar $picard MarkDuplicates \
INPUT=sorted_reads.bam \
OUTPUT=dedup_reads.bam \
METRICS_FILE=metrics.txt

## Index bam file
$java -jar $picard BuildBamIndex \
INPUT=dedup_reads.bam
\end{lstlisting}

\section{Base Quality Recalibration}
\begin{lstlisting}[language=bash]
## Analyze patterns of covariation in the sequence dataset
$java -jar $gatk \
-T BaseRecalibrator \
-R $assembly \
-nct 8 \
-I dedup_reads.bam \
-knownSites $known_snp \
-knownSites $known_indels \
-L $regions \
-o recal_data.table

## Do a second pass to analyze covariation remaining after recalibration
$java -jar $gatk \
-T BaseRecalibrator \
-R $assembly \
-nct 8 \
-I dedup_reads.bam \
-knownSites $known_snp \
-knownSites $known_indels \
-BQSR recal_data.table \
-L $regions \
-o post_recal_data.table

## Apply recalibration
$java -jar $gatk \
-T PrintReads \
-R $assembly \
-nct 8 \
-I dedup_reads.bam \
-BQSR recal_data.table \
-o recal_reads.bam
\end{lstlisting}

\section{Germline Variant Calling}
\begin{lstlisting}[language=bash]
$java -Xmx10g -jar $gatk \
--analysis_type HaplotypeCaller \
-nct 8 \
--reference_sequence $assembly \
-I $inNormal/recal_reads.bam \
-o $inNormal/variants.vcf \
-L $regions \
--dbsnp $known_snp
\end{lstlisting}

\section{Somatic Variant Calling for Tumors}
\begin{lstlisting}[language=bash]
$java -Xmx10g -jar $gatk \
--analysis_type MuTect2 \
--reference_sequence $assembly \
--input_file:normal $inNormal/recal_reads.bam \
--input_file:tumor $inTumor/recal_reads.bam \
--out $inTumor/somatic_variants.vcf \
--cosmic $cosmic \
--dbsnp $known_snp \
-L $regions \
-nct 8
\end{lstlisting}

\section{Annotation of variants using Oncotator}
\begin{lstlisting}[language=bash]
## Filter out variants with PASS
/data/apps/vcftools_0.1.13/bin/vcftools \
--vcf somatic_variants.vcf \
--remove-filtered-all \
--out somatic_variants.filtered \
--recode

## Start Virtual Machine
source /data/users/rasmus/software/oncotator_vm_1.9/bin/activate

## Run oncotator
/data/users/rasmus/software/oncotator_vm_1.9/bin/oncotator \
-i VCF \
-o TCGAMAF \
--db-dir /data/appdata/oncotator_v1_ds_Jan262014/ \
somatic_variants.filtered.recode.vcf \
somatic_variants_filtered_oncotator.maf \
hg19

\end{lstlisting}
\end{document}